File: backend/src/main/java/by/fpmibsu/findafriend/controller/controllers/PlaceController.java
package by.fpmibsu.findafriend.controller.controllers;

import by.fpmibsu.findafriend.application.HandleResult;
import by.fpmibsu.findafriend.application.controller.*;
import by.fpmibsu.findafriend.application.mediatr.Mediatr;
import by.fpmibsu.findafriend.controller.Logging;
import by.fpmibsu.findafriend.controller.commands.places.CreatePlaceCommand;
import by.fpmibsu.findafriend.controller.commands.places.DeletePlaceCommand;
import by.fpmibsu.findafriend.controller.commands.places.UpdatePlaceCommand;
import by.fpmibsu.findafriend.controller.queries.places.GetPlaceByIdQuery;
import by.fpmibsu.findafriend.controller.queries.places.GetPlacesQuery;
import by.fpmibsu.findafriend.entity.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@ControllerRoute(route = "/places")
public class PlaceController extends Controller {
    private static final Logger logger = LogManager.getLogger(PlaceController.class);
    private final Mediatr mediatr;

    public PlaceController(Mediatr mediatr) {
        this.mediatr = mediatr;
    }

    @Endpoint(path = "/getAll", method = HttpMethod.GET)
    public HandleResult getAll() throws Exception {
        return ok(mediatr.send(new GetPlacesQuery()));
    }

    @Endpoint(path = "/getById", method = HttpMethod.GET)
    public HandleResult getById(@FromQuery(parameterName = "id") int id) throws Exception {
        return ok(mediatr.send(new GetPlaceByIdQuery(id)));
    }

    @RequireAuthentication
    @Endpoint(path = "/create", method = HttpMethod.POST)
    public HandleResult createPlace(@FromBody CreatePlaceCommand request, @WebToken(parameterName = "role") String role) throws Exception {
        if (!User.Role.ADMINISTRATOR.toString().equals(role)) {
            Logging.warnNonAuthorizedAccess(this.request, logger);
            return notAuthorized();
        }
        return ok(mediatr.send(request));
    }

    @RequireAuthentication
    @Endpoint(path = "/update", method = HttpMethod.PUT)
    public HandleResult updatePlace(@FromBody UpdatePlaceCommand request, @WebToken(parameterName = "role") String role) throws Exception {
        if (!User.Role.ADMINISTRATOR.toString().equals(role)) {
            Logging.warnNonAuthorizedAccess(this.request, logger);
            return notAuthorized();
        }
        return ok(mediatr.send(request));
    }

    @RequireAuthentication
    @Endpoint(path = "/delete", method = HttpMethod.DELETE)
    public HandleResult deletePlaceById(@FromQuery(parameterName = "id") int id, @WebToken(parameterName = "role") String role) throws Exception {
        if (!User.Role.ADMINISTRATOR.toString().equals(role)) {
            Logging.warnNonAuthorizedAccess(this.request, logger);
            return notAuthorized();
        }
        return ok(mediatr.send(new DeletePlaceCommand(id)));
    }
}

File: BackendNET/Backend/Controllers/PlaceController.cs
using Backend.Commands.Places.CreatePlace;
using Backend.Commands.Places.DeletePlace;
using Backend.Commands.Places.UpdatePlace;
using Backend.Dto;
using Backend.Queries.Places.GetAll;
using Backend.Queries.Places.GetById;
using MediatR;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Backend.Controllers;

public class PlaceController : ControllerBase
{
    private readonly IMediator _mediator;

    public PlaceController(IMediator mediator)
    {
        _mediator = mediator;
    }

    [HttpGet("/places/getAll")]
    public async Task<ActionResult<IEnumerable<PlaceDto>>> GetAllAsync()
    {
        return Ok(await _mediator.Send(new GetAllPlacesQuery()));
    }

    [HttpGet("/places/getById")]
    public async Task<ActionResult<PlaceDto?>> GetByIdAsync([FromQuery] int id)
    {
        return Ok(await _mediator.Send(new GetPlaceByIdQuery { PlaceId = id }));
    }

    [Authorize(Roles = "Administrator")]
    [HttpPost("/places/create")]
    public async Task<IActionResult> CreateAsync([FromBody] CreatePlaceCommand command)
    {
        return Ok(await _mediator.Send(command));
    }

    [Authorize(Roles = "Administrator")]
    [HttpPut("/places/update")]
    public async Task<IActionResult> UpdateAsync([FromBody] UpdatePlaceCommand command)
    {
        return Ok(await _mediator.Send(command));
    }

    [Authorize(Roles = "Administrator")]
    [HttpDelete("/places/delete")]
    public async Task<IActionResult> DeleteAsync([FromQuery] int id)
    {
        return Ok(await _mediator.Send(new DeletePlaceCommand { PlaceId = id }));
    }
}